plugins/techdocs-addons/src/reader.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CompoundEntityRef } from '@backstage/catalog-model';
+import { Content, Header, Page, Progress } from '@backstage/core-components';
+import { configApiRef, useApi } from '@backstage/core-plugin-api';
+// todo(backstage/techdocs-core): Export these from @backstage/plugin-techdocs
+import {
+  // @ts-ignore
+  useTechDocsReaderDom,
+  // @ts-ignore
+  withTechDocsReaderProvider,
+  // @ts-ignore
+  TechDocsStateIndicator as TechDocReaderPageIndicator,
+} from '@backstage/plugin-techdocs';
+import {
+  withStyles,
+  Portal,
+  Box,
+  Toolbar,
+  ToolbarProps,
+} from '@material-ui/core';
+import { Skeleton } from '@material-ui/lab';
+import { StylesProvider, jssPreset } from '@material-ui/styles';
+import React, { useEffect, useRef, useState } from 'react';
+import { create } from 'jss';
+import Helmet from 'react-helmet';
+
+import { useTechDocsAddons } from './addons';
+import {
+  TechDocsMetadataProvider,
+  useMetadata,
+  TechDocsEntityProvider,
+  TechDocsReaderPageProvider,
+  useTechDocsReaderPage,
+} from './context';
+import { TechDocsAddonLocations as locations } from './types';
+
+const TechDocsReaderPageSubheader = withStyles(theme => ({
```

Comment:
what do you think about separate these components into its own files? for improved readability?